stocks2.lua

```diff
@@ -0,0 +1,544 @@
+                        {text = "Shift-T", pen = COLOR_LIGHTGREEN}, {text = ": Mark for Trade", pen = COLOR_GREY},
+                    },
+                    frame = {t = 17, l = 0},
+                    view_id = "actions_label",
+                }),
+                widgets.Label({
+                    text = {
+                        {text = "Shift-A", pen = COLOR_LIGHTGREEN}, {text = ": Apply to: ", pen = COLOR_WHITE},
+                        {text = apply_to_labels[apply_to_mode + 1], pen = COLOR_BROWN},
+                    },
+                    frame = {t = 20, l = 0},
+                    view_id = "apply_to_label",
+                }),
+            }
+        }),
+    })
+
+    self:populate_item_table()
+    self:filter_list(current_search_query)
+    self:show_filters()
+    self:show_qual_and_wear()
+end
+
+function StocksWindow:populate_item_table()
+    -- retrieve all in-play items
+    local temp_table = {}
+    for i,v in ipairs(df.global.world.items.other[df.items_other_id.IN_PLAY]) do
+        local item_desc = dfhack.items.getDescription(v, v:getType())
+
+        -- store item description and an item ref
+        table.insert(temp_table, { desc = item_desc, ref = v })
+    end
+
+    -- sort alphabetically
+    local spec = { key = function(v) return v.desc end }
+    local order = utils.make_sort_order(temp_table, { spec })
+    for i = 1, #order do item_table[i] = temp_table[order[i]] end
+end
+
+function toggle_melt(item)
+    item.flags.melt = not item.flags.melt
+
+    if item.flags.melt then
+        utils.insert_sorted(df.global.world.items.other.ANY_MELT_DESIGNATED, item, "id")
+    else
+        utils.erase_sorted(df.global.world.items.other.ANY_MELT_DESIGNATED, item, "id")
+    end
+end
+
+function is_valid_melt_item(item)
+    local mat_info = dfhack.matinfo.decode(item)
+    if not mat_info then
+        return false
+    end
+
+    -- must be a metal thing
+    if mat_info:getCraftClass() ~= df.craft_material_class.Metal then
+        return false
+    end
+
+    -- must not be a metal bar
+    if item:getType() == df.item_type.BAR then
+        return false
+    end
+
+    for i,g in ipairs(item.general_refs) do
+        local t = g:getType()
+
+        -- don't be a holder of things
+        if t == df.general_ref_type.CONTAINS_ITEM or
+           t == df.general_ref_type.UNIT_HOLDER or
+           t == df.general_ref_type.CONTAINS_UNIT then
+           return false
+        end
+
+        -- or be in something like a quiver, backpack, flask etc?
+        if t == df.general_ref_type.CONTAINED_IN_ITEM then
+            for i2, g2 in ipairs(g:getItem().general_refs) do
+                if g2:getType() == df.general_ref_type.UNIT_HOLDER then
+                    return false
+                end
+            end
+        end
+    end
+
+    -- lastly, don't melt masterworks
+    if item:getQuality() >= 5 then
+        return false
+    end
+
+    return true
+end
+
+function StocksWindow:onInput(keys)
```

Comment:
once appropriate mouse-aware widgets are being used, I believe you can get rid of this function entirely